### ISSUE TITLE
Support simulation of display rotation

### ISSUE THREAD

**Issue by jaustin** [microsoft/pxt-microbit#4935]
With the [pxt-rotate](https://github.com/bsiever/microbit-pxt-rotate) extension, it's possible to create nice examples that show how mobile phone accelerometer is used to update the display rotation based on device orientation.

This has been requested by micro:bit teachers in Slack (which is why the extension was created) and it's hard to do with the icons and text otherwise.

However, right now, the CODAL feature for display rotation isn't supported on the simulator.

It would be great to have a design for this. Initially it could just be that the display rotates simply, but as it is most likely related to a kind of micro:bit motion that we can't do on the sim we may want to consider wider changes.

See also:
https://github.com/lancaster-university/codal-microbit-v2/pull/227

CC @bsiever @finneyj

**Comment by bsiever** [comment]
The extension is trivial.  If the simulator supports rotation too, the blocks might as well be included in the default toolboxes (and the extension deprecated/removed).   The extension still depends on CODAL updates which are contingent on the next annual release.  If possible, maybe target simulator support and new rotation blocks for the release?

**Comment by jaustin** [comment]
Thanks @bsiever that was the reason I created the issue, to consider putting this into the 2023 release :)

I agree too about pulling the extension into the core code, but I think that should be a separate issue  - it'd avoid the long compile times for the small C++ addition. @pelikhan @abchatra any thoughts?
